brilliant-builders/firestarter/game_engine/engine.py

```diff
@@ -0,0 +1,173 @@
+from itertools import chain
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+from firestarter.game_engine.resources_loader import load_resources
+from firestarter.game_engine.sprite import Sprite
+from firestarter.game_engine.utils import get_all_subclasses
+
+from kivy.clock import Clock
+from kivy.core.window import Keyboard, Window
+from kivy.logger import Logger
+from kivy.uix.widget import Widget
+
+IMAGE_EXTENSIONS = ['png']
+
+
+class Engine(Widget):
+    resource_dir = (Path('.') / 'firestarter' / 'resources').absolute()
```

Comment:
pathlib's `__div__` operation is pretty handy, but you don't necessarily need it in this case. This line could be shortened to `resource_dir = Path('firestarter/resources')`
This pattern is repeated in `resources_loader.py` and `sprite.py`: It might have been worth making a `constants.py` file and importing from there:
```python
class Paths:
    RESOURCES = Path('firestarter/resources')
...
from constants import Paths
...
class Engine(Widget):
    resource_dir = Paths.RESOURCES
```